format_checker/common_checks.py

```diff
@@ -123,6 +146,9 @@ def run_checks(file, data_dict, log, commit_range, checks):
                         if check_rule.__name__ == check_row_length.__name__:
                             check_rule(len(header), *params)
                             continue
+                        if check_rule.__name__ == check_repo_sanity.__name__:
```

Comment:
Instead of running this check_rule for every changed line in the CSV file, we could just run this check when a new Project URL is added that does not already exist. i.e., get (1) the Project URLs that are already in the unchanged file and (2) the Project URLs in the new changes, and only run this check for Project URLs that are in (2) but not in (1).

This improvement need not be done immediately but would help avoid the Github limit.